### PR TITLE
wavebox: 10.114.26-2 -> 10.118.5-2

### LINKED ISSUE #254908: Update request: wavebox 10.114.26-2 → 10.116.10-2
- Package name: wavebox
- Latest released version: [10.116.10-2](https://github.com/wavebox/waveboxapp/releases/latest)
- Current version on the unstable channel: [10.114.26-2](https://search.nixos.org/packages?channel=unstable&show=wavebox&from=0&size=50&sort=relevance&type=packages&query=wavebox)
- Current version on the stable/release channel: [4.11.3](https://search.nixos.org/packages?channel=23.05&show=wavebox&from=0&size=50&sort=relevance&type=packages&query=wavebox)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+is%3Aopen+wavebox) (doesn't seem like the [Watir2 PR](https://github.com/NixOS/nixpkgs/pull/115501) has anything to do with Wavebox 114 to 116?)

**Notify maintainers**

@rawkode @eddsteel @cole-h

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by FliegendeWurst**
## Description of changes

Update to latest version. It seems there is now an optional dependency on Qt 6 (Qt 5 is still required).

Fixes #254908

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bayazidbh** [comment]
Latest version is now 10.117.21-2. Is there a possibility of having the version be tracked automatically since I think the main changes required for the nixpkgs source is to update the version number and SHA256 value?

**Comment by FliegendeWurst** [comment]
> Is there a possibility of having the version be tracked automatically since I think the main changes required for the nixpkgs source is to update the version number and SHA256 value?

Yes, that is possible with an update script. For that we will need some API to determine the latest version.

**Comment by Thomas101** [comment]
@FliegendeWurst you can use `https://download.wavebox.app/stable/linux/latest.json` to get the latest version 👍

**Comment by FliegendeWurst** [comment]
> @FliegendeWurst you can use `https://download.wavebox.app/stable/linux/latest.json` to get the latest version 👍 

Thank you for the info. Seems like a better way than what I came up with (parsing the filename of the latest download URL). Will update the script later.

File: pkgs/applications/networking/instant-messengers/wavebox/default.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@
 }:
 
 let
-  version = "10.114.26-2";
+  version = "10.117.21-2";
```
Comment (Artturin) [inline pkgs/applications/networking/instant-messengers/wavebox/default.nix:21]: this attr can be moved out of the let once the unneeded tarball attr is removed.

File: pkgs/applications/networking/instant-messengers/wavebox/update.sh
Comment (Artturin) [inline pkgs/applications/networking/instant-messengers/wavebox/update.sh:1]: You should be able to simplify this by using `nix-update`
Example https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/abcl/update.sh

Comment (FliegendeWurst) [reply pkgs/applications/networking/instant-messengers/wavebox/update.sh:1]: Thank you for the suggestion. Using nix-update is much easier than rewriting the hash manually!

**Pull request merged**